src/bigDecimal.ts

```diff
@@ -11,14 +11,26 @@ function isNumericValue(n: any): n is number {
 
 export function fromBigDec(bigDec: BigDecimal) {
   const { scale, value } = bigDec;
-  if (!isNumericValue(scale) || !isNumericValue(value)) {
```

Comment:
I think this check should be better done before detecting whether scale is between 0 and 100.